### PR TITLE
Fix mongodb instalation

### PR DESCRIPTION
**Pull request by TStypulkowski**

### CODE REVIEW

**Review by patkaczm — CHANGES_REQUESTED** [review]
Seems that this branch hasn't been `rebased`. Please do `rebase` first.

File: Dockerfile [hunk 1]
```diff
@@ -56,19 +56,19 @@ RUN apt-get install -y \
     libssl-dev \
     libsasl2-dev \
     libbson-dev \
-    git \
-    curl
```
Comment (ddnowicki) [inline Dockerfile:60]: we use curl so I think it should stay

File: src/database/database.cpp [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
+  DatabaseImpl(const std::string& address);
+  ~DatabaseImpl() = default;
+
+  bool isThereUserWithThisUsername(const std::string& username) const;
+  bool isThereUserWithThisEmail(const std::string& email) const;
+  bool createUser(const std::string& username, const std::string& password,
+                  const std::string& email);
+  bool deleteUser(const std::string& id, const std::string& password);
+  bool loginCheck(const std::string& id, const std::string& password) const;
+};
```
Comment (ddnowicki) [inline src/database/database.cpp]: `#include` and class implementation should be in hpp file

**Review by ddnowicki — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (TStypulkowski) [reply Dockerfile:60]: yes, but it is still in dockerfile. This one was second(they were doubled) one, so I removed it

File: src/database/database.hpp [hunk 3]
```diff
@@ -1,35 +1,20 @@
...
-  bool isThereUserWithThisEmail(const std::string& email) const;
-  bool isUserPasswordEqualGivenPassword(mongocxx::cursor& cursor,
-                                        const std::string& password) const;
   bool createUser(const std::string& username, const std::string& password,
                   const std::string& email);
   bool deleteUser(const std::string& id, const std::string& password);
   bool loginCheck(const std::string& id, const std::string& password) const;
+
+  private:
+  std::unique_ptr<DatabaseImpl> impl;
```
Comment (patkaczm) [inline src/database/database.hpp:19]: Why pimpl?

Comment (TStypulkowski) [reply src/database/database.hpp:19]: database-lib is not included in CMakeList in core and api, and therefore there were a lot of errors. With pimpl database-lib doesn't have to be in CMakeList in core and api, and everything works just fine. Another reason is that now there will be no errors no matter where others include database.hpp.

**Review by patkaczm — APPROVED** [review]

**Review by ddnowicki — APPROVED** [review]

**Pull request merged**